frontend/packages/metal3-plugin/src/types/misc.ts

```diff
@@ -0,0 +1,6 @@
+import { K8sResourceCommon, K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResult } from '@console/internal/components/utils';
+
+export type OptionalFirehoseResult<
```

Comment:
Maybe this could be put somewhere near the FirehoseResult definition instead of metal3-plugin?

Reply:
#4016